src/sentry/integrations/slack/views/unlink_team.py

```diff
@@ -88,12 +87,22 @@ def handle(self, request: Request, signed_params: str) -> HttpResponse:
             external_id=channel_id,
         )
         if len(external_teams) == 0:
-            return render_error_page(request, body_text="HTTP 404: Team not found")
-
-        if external_teams[0].team_id is None:
-            return render_error_page(request, body_text="HTTP 404: Team not found")
+            return render_error_page(request, status=404, body_text="HTTP 404: Team not found")
 
         team = external_teams[0].team
+        if team is None:
+            return render_error_page(request, status=404, body_text="HTTP 404: Team not found")
+
+        # Error if you don't have a sufficient role and you're not a team admin
+        # on the team you're trying to unlink.
+        if not is_valid_role(om) and not is_team_admin(om, team=team):
+            logger.info(
+                "slack.action.invalid-role",
+                extra={"slack_id": integration.external_id, "user_id": request.user.id},
+            )
+            return render_error_page(
+                request, status=404, body_text="HTTP 404: " + INSUFFICIENT_ACCESS
+            )
```

Comment:
Instead of only returning a blank 404, we now return the actual error page with a message on it.